app/controllers/foreman_ansible/concerns/api_common.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module ForemanAnsible
+  module Concerns
+    # Shared task methods between api controllers
+    module ApiCommon
+      extend ActiveSupport::Concern
+
+      def find_ansible_roles
+        ids = params.fetch(:ansible_role_ids, []) || []
+        @ansible_roles = []
```

Comment:
This way is not very efficient, it loads each role with separate SQL query, also the current way does not authorize resources. How about this single line?

```
AnsibleRole.authorized(:view_ansible_roles).find(ids)
```

it also raises ActiveRecord::RecordNotFound in case some of the id does not exist, so the rescue part could remain the same